docs/libcurl/opts/CURLOPT_SAFE_AUTH.3

```diff
@@ -0,0 +1,59 @@
+.\" **************************************************************************
+.\" *                                  _   _ ____  _
+.\" *  Project                     ___| | | |  _ \| |
+.\" *                             / __| | | | |_) | |
+.\" *                            | (__| |_| |  _ <| |___
+.\" *                             \___|\___/|_| \_\_____|
+.\" *
+.\" * Copyright (C) 1998 - 2022, Daniel Stenberg, <daniel@haxx.se>, et al.
+.\" *
+.\" * This software is licensed as described in the file COPYING, which
+.\" * you should have received as part of this distribution. The terms
+.\" * are also available at https://curl.se/docs/copyright.html.
+.\" *
+.\" * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+.\" * copies of the Software, and permit persons to whom the Software is
+.\" * furnished to do so, under the terms of the COPYING file.
+.\" *
+.\" * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+.\" * KIND, either express or implied.
+.\" *
+.\" **************************************************************************
+.\"
+.TH CURLOPT_SAFE_AUTH 3 "18 Jan 2022" "libcurl 7.82.0" "curl_easy_setopt options"
+.SH NAME
+CURLOPT_SAFE_AUTH \- do not use clear password authentication
+.SH SYNOPSIS
+.nf
+#include <curl/curl.h>
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_SAFE_AUTH, long onoff);
+.fi
+.SH DESCRIPTION
+Pass a long as parameter, 0 or 1 where 1 is for enable and 0 for disable.
+When set, This option rejects authentication mechanisms that would
+transfer clear passwords on a non-encrypted connection.
```

Comment:
It feels reasonable to expect a user to mostly want to set this for both host+proxy at the same time, or not at all. Therefore I think it could make sense to be able to set both using the same libcurl option using a bitmask instead of adding two separate options.

Reply:
What happens if this option is set true and `CURLOPT_HTTPAUTH` sets Basic? It would probably help if this spells that out.

Reply:
Thanks for the review.
Sure, these are contradictory options.
In case it occurs, the CURLOPT_SAFE_AUTH takes the precedence. This is also true for SASL mechanisms. I'll add a doc paragraph for that.

Reply:
> ... set both using the same libcurl option using a bitmask instead of adding two separate options.

I did it that way because we don't have a `curl_easy_getopt()` (BTW: possible future feature?). But I'll change it if you prefer.
What about flag names? I suggest CURLSAFE_PWD and CURLSAFE_PROXY_PWD.